output.tf

```
@@ -1,11 +1,7 @@
-output "security_group_id" {
-  value = aws_security_group.elasticache.id
-}
-
-output "endpoints" {
-  value = aws_elasticache_cluster.default.cache_nodes.*.address
+output "primary_endpoints" {
+  value = var.replication_group ? aws_elasticache_replication_group.default[0].primary_endpoint_address : aws_elasticache_cluster.default[0].cache_nodes.*.address
```

Maybe this reads a bit easier:
```suggestion
  value = try(aws_elasticache_cluster.default[0].cache_nodes.*.address, aws_elasticache_replication_group.default[0].primary_endpoint_address)
```